src/plugins/hetero/src/plugin.cpp

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+#include "itt.hpp"
+#include "plugin.hpp"
+#include "compiled_model.hpp"
+
+#include <memory>
+#include <vector>
+#include <map>
+#include <string>
+#include <utility>
+#include <fstream>
+#include <unordered_set>
+
+#include "openvino/runtime/device_id_parser.hpp"
+#include "openvino/runtime/properties.hpp"
+#include "openvino/runtime/internal_properties.hpp"
+#include "openvino/util/common_util.hpp"
+#include "ie/ie_plugin_config.hpp"
+
+#include "hetero/properties.hpp"
+
+ov::hetero::Plugin::Plugin() {
+    set_device_name("HETERO");
+}
+
+std::shared_ptr<ov::ICompiledModel> ov::hetero::Plugin::compile_model(
+    const std::shared_ptr<const ov::Model>& model,
+    const ov::AnyMap& properties) const {
+    OV_ITT_SCOPED_TASK(itt::domains::Hetero, "Plugin::compile_model");
+    
+    auto config = Configuration{properties, m_cfg};
+    auto compiled_model = std::make_shared<CompiledModel>(
+        model->clone(),
+        shared_from_this(),
+        config);
+    return compiled_model;
+}
+
+std::shared_ptr<ov::ICompiledModel> ov::hetero::Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+    const ov::AnyMap& properties,
+    const ov::RemoteContext& context) const {
```

Comment:
@ilyachur looks strange that `context` is last argument here, while in the method below `context` is earlier than `properties`.